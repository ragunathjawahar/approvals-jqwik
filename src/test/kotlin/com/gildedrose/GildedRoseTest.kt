package com.gildedrose

import net.jqwik.api.Arbitraries
import net.jqwik.api.Arbitrary
import net.jqwik.api.ForAll
import net.jqwik.api.From
import net.jqwik.api.Property
import net.jqwik.api.Provide
import org.approvaltests.Approvals

class GildedRoseTest {
  companion object {
    const val DATA_SET_SIZE = 50
  }

  @Property(
    tries = 1,
    seed = "-8556729318537930054"
  )
  fun `Aged Brie`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Aged Brie"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "1002985723690070170"
  )
  fun `Backstage passes to a TAFKAL80ETC concert`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Backstage passes to a TAFKAL80ETC concert"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "3398327011427988732"
  )
  fun `Sulfuras, Hand of Ragnaros`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Sulfuras, Hand of Ragnaros"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "6350692656906137395"
  )
  fun `all other items`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "all other items"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Provide
  fun sellIns(): Arbitrary<List<Int>> =
    Arbitraries
      .integers()
      .between(0, 50)
      .list()
      .uniqueElements()
      .ofSize(DATA_SET_SIZE)

  @Provide
  fun qualities(): Arbitrary<List<Int>> =
    Arbitraries
      .integers()
      .between(-10, 60)
      .list()
      .uniqueElements()
      .ofSize(DATA_SET_SIZE)
}
