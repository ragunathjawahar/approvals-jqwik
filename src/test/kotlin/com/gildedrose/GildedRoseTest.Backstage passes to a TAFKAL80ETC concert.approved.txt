Backstage passes to a TAFKAL80ETC concert, 49, 59
Backstage passes to a TAFKAL80ETC concert, 4, 19
Backstage passes to a TAFKAL80ETC concert, 19, 16
Backstage passes to a TAFKAL80ETC concert, 2, 4
Backstage passes to a TAFKAL80ETC concert, 30, -1
Backstage passes to a TAFKAL80ETC concert, 15, 30
Backstage passes to a TAFKAL80ETC concert, 27, -4
Backstage passes to a TAFKAL80ETC concert, 5, 34
Backstage passes to a TAFKAL80ETC concert, 26, 18
Backstage passes to a TAFKAL80ETC concert, 43, -7
Backstage passes to a TAFKAL80ETC concert, 47, 1
Backstage passes to a TAFKAL80ETC concert, -1, 0
Backstage passes to a TAFKAL80ETC concert, 13, -8
Backstage passes to a TAFKAL80ETC concert, 9, 28
Backstage passes to a TAFKAL80ETC concert, 25, 46
Backstage passes to a TAFKAL80ETC concert, 46, -5
Backstage passes to a TAFKAL80ETC concert, 38, 11
Backstage passes to a TAFKAL80ETC concert, 11, 25
Backstage passes to a TAFKAL80ETC concert, 10, -3
Backstage passes to a TAFKAL80ETC concert, 28, 60
Backstage passes to a TAFKAL80ETC concert, 8, -8
Backstage passes to a TAFKAL80ETC concert, 39, -2
Backstage passes to a TAFKAL80ETC concert, 48, 32
Backstage passes to a TAFKAL80ETC concert, 20, 14
Backstage passes to a TAFKAL80ETC concert, 12, 35
Backstage passes to a TAFKAL80ETC concert, 7, 13
Backstage passes to a TAFKAL80ETC concert, 35, 20
Backstage passes to a TAFKAL80ETC concert, 14, 53
Backstage passes to a TAFKAL80ETC concert, 44, 9
Backstage passes to a TAFKAL80ETC concert, 31, 3
Backstage passes to a TAFKAL80ETC concert, 34, -6
Backstage passes to a TAFKAL80ETC concert, 1, 25
Backstage passes to a TAFKAL80ETC concert, 6, 20
Backstage passes to a TAFKAL80ETC concert, 22, 54
Backstage passes to a TAFKAL80ETC concert, 24, 4
Backstage passes to a TAFKAL80ETC concert, 33, 26
Backstage passes to a TAFKAL80ETC concert, 32, 10
Backstage passes to a TAFKAL80ETC concert, 0, 17
Backstage passes to a TAFKAL80ETC concert, 18, 58
Backstage passes to a TAFKAL80ETC concert, 23, 47
Backstage passes to a TAFKAL80ETC concert, 41, 56
Backstage passes to a TAFKAL80ETC concert, 21, 5
Backstage passes to a TAFKAL80ETC concert, 36, 52
Backstage passes to a TAFKAL80ETC concert, 17, 31
Backstage passes to a TAFKAL80ETC concert, 3, 50
Backstage passes to a TAFKAL80ETC concert, 42, 51
Backstage passes to a TAFKAL80ETC concert, 37, 37
Backstage passes to a TAFKAL80ETC concert, 45, 40
Backstage passes to a TAFKAL80ETC concert, 29, 43
Backstage passes to a TAFKAL80ETC concert, 40, 48
